### PR TITLE
Add WEB-support for CoreStoreSembastImp

### PR DESCRIPTION
**Pull request by maaeps**
This is pretty much the same PR as @nstrelow's  #394.

Please have a look at  #394 for a detailed explanation of the problem.

I think, although it's not optimal,  not encrypting the database is the best trade we can make.

As far as I know, this should add full web support to the SDK. (#385)

### CODE REVIEW

**Comment by phillwiggins** [comment]
Would this still allow encryption using Flutter but not Dart?

**Comment by phillwiggins** [comment]
There are conflicts with this branch now.

**Comment by nstrelow** [comment]
@phillwiggins 
The opposite right? Encryption as long as it is not running on web.
So Android, iOS and native dart probably.

@maaeps Maybe print a debug warning when running on web and using Sembast

**Comment by maaeps** [comment]
All conflicts should be properly fixed now.

@nstrelow 
Great idea!

File: lib/src/storage/core_store_sem_impl.dart [hunk 1]
```diff
@@ -14,8 +14,16 @@ class CoreStoreSembastImp implements CoreStore {
       if (!parseIsWeb &&
           (Platform.isIOS || Platform.isAndroid || Platform.isMacOS))
         dbDirectory = (await getApplicationDocumentsDirectory()).path;
+      assert(() {
+        if (parseIsWeb) {
+          print(
+              'Warning: CoreStoreSembastImp of the Parse_Server_SDK does not encrypt the database on WEB.');
```
Comment (nstrelow) [inline lib/src/storage/core_store_sem_impl.dart:16]: Awesome :D

**Review by nstrelow — APPROVED** [review]

File: lib/src/storage/core_store_sem_impl.dart (hunk 1 above)
Comment (maaeps) [reply lib/src/storage/core_store_sem_impl.dart:16]: I found this on stackoverflow.
I have to remember this trick, it's super useful.

**Pull request merged**